### PR TITLE
Bump Megaparsec to 7.0.4

### PR DESCRIPTION
**Pull request by vaibhavsagar**

### CODE REVIEW

**Comment by vaibhavsagar** [comment]
This doesn't build yet and it looks like all the fun features we used from `Text.Megaparsec.Error` have been changed for the worse now.

**Comment by vaibhavsagar** [comment]
This actually compiles now and all the tests pass, which is strange.

**Comment by vaibhavsagar** [comment]
The error output has changed. This is from `master`:

```bash
$ dist/build/hnix/hnix -E "let a = b in c"
hnix: Parse failed: <string>:1:11:
  |
1 | let a = b in c
  |           ^
unexpected 'i'
expecting ';', ., operator, or the rest of variable binding
```

and this is what it looks like after these changes:

```bash
$ dist/build/hnix/hnix -E "let a = b in c"
hnix: Parse failed: <string>:1:11:
  |
1 | let a = b in c
  |           ^
unexpected 'i'
expecting "!=", "&&", "++", "->", "//", "<=", "==", ">=", "||", '*', '+', '-', '/', ';', '<', '>', '?', or .

```

File: src/Nix/Render.hs [hunk 1]
```diff
@@ -10,26 +10,22 @@
...
 
-renderLocation :: MonadFile m => SrcSpan -> Doc -> m Doc
-renderLocation (SrcSpan beg@(SourcePos "<string>" _ _) _) msg =
-    return $ text $ init $ parseErrorPretty @Char (posAndMsg beg msg)
-
-renderLocation (SrcSpan beg@(SourcePos path _ _) _) msg = do
-    contents <- Nix.Render.readFile path
-    return $ text $ init $ parseErrorPretty' contents (posAndMsg beg msg)
+renderLocation :: Monad m => SrcSpan -> Doc -> m Doc
+renderLocation (SrcSpan beg@(SourcePos _ _ _) _) msg =
```
Comment (domenkozar) [inline src/Nix/Render.hs:74]: I think we still need to handle `<string>` case

**Comment by domenkozar** [comment]
One issue and merge conflicts and I think we're good to go :)

File: src/Nix/Render.hs (hunk 1 above)
Comment (vaibhavsagar) [reply src/Nix/Render.hs:74]: How should it be handled differently?

Comment (domenkozar) [reply src/Nix/Render.hs:74]: Ah, seems like path doesn't need resolving anymore and that was the special case. All good :)

**Comment by vaibhavsagar** [comment]
Should be good now.

**Comment by jwiegley** [comment]
@vaibhavsagar Can you resolve the current conflicts and repush? I'd love to merge this in.

**Comment by vaibhavsagar** [comment]
Sure, updated. I'm still building dependencies so I haven't tested this locally just yet but I will repush again if it turns out I broke something when resolving conflicts.

**Comment by vaibhavsagar** [comment]
Confirmed that this successfully builds locally.

**Comment by vaibhavsagar** [comment]
Thanks!

**Pull request merged**